src/Compilers/CSharp/Portable/Symbols/TypeSymbolExtensions.cs

```diff
@@ -2078,6 +2078,12 @@ internal static bool IsWellKnownTypeIsExternalInit(this TypeSymbol typeSymbol)
 
         internal static bool IsWellKnownTypeOutAttribute(this TypeSymbol typeSymbol) => typeSymbol.IsWellKnownInteropServicesTopLevelType("OutAttribute");
 
+        internal static bool IsWellKnownTypeLock(this TypeSymbol typeSymbol)
+        {
+            return typeSymbol.Name == "Lock" && typeSymbol.ContainingType is null &&
+                typeSymbol.IsContainedInNamespace("System", "Threading");
```

Comment:
This isn't how we usually check for a ```WellKnownTipe``` especially if we are planning to rely on a ```WellKnownMember``` for it.